### DISCUSSION TITLE
i can't open friday night funkin with kade engine

### DISCUSSION THREAD

**Discussion by RuvIceCream** [KadeArchive/Kade-Engine#1140] [Q&A]
Can you explain to me why when I do lime test windows I get this error?source/Discord.hx:5: characters 8-30 : Type not found : discord_rpc.DiscordRpc
source/Discord.hx:14: characters 3-13 : Type not found : DiscordRpc
source/Discord.hx:24: characters 4-14 : Type not found : DiscordRpc
source/Discord.hx:29: characters 3-13 : Type not found : DiscordRpc
source/Discord.hx:34: characters 3-13 : Type not found : DiscordRpc
source/Discord.hx:39: characters 3-13 : Type not found : DiscordRpc
source/Discord.hx:75: characters 3-13 : Type not found : DiscordRpc
C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189: characters 10-23 : Warning : This typedef is deprecated in favor of haxe.xml.Access
C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189: characters 30-43 : Warning : This typedef is deprecated in favor of haxe.xml.Access

someone know how fix?

**Comment by luRaichu** [comment]
Are you on Mac?

**Reply by RuvIceCream** [reply]
no

**Comment by EcoVeni** [comment]
Did you do command "haxelib install discord-rpc" ?

**Reply by RuvIceCream** [reply]
yes but nothing

**Reply by EcoVeni** [reply]
then that's weird. idk how to fix it beacuse i never got it

**Reply by EcoVeni** [reply]
maybe copy this code and paste it to the Discord.hx file

**Reply by EcoVeni** [reply]
package;

#if windows
import Sys.sleep;
import discord_rpc.DiscordRpc;

using StringTools;

class DiscordClient
{
	public function new()
	{
		trace("Discord Client starting...");
		DiscordRpc.start({
			clientID: "557069829501091850", // change this to what ever the fuck you want lol
			onReady: onReady,
			onError: onError,
			onDisconnected: onDisconnected
		});
		trace("Discord Client started.");

		while (true)
		{
			DiscordRpc.process();
			sleep(2);
			//trace("Discord Client Update");
		}

		DiscordRpc.shutdown();
	}

	public static function shutdown()
	{
		DiscordRpc.shutdown();
	}

	static function onReady()
	{
		DiscordRpc.presence({
			details: "In the Menus",
			state: null,
			largeImageKey: 'icon',
			largeImageText: "fridaynightfunkin"
		});
	}

	static function onError(_code:Int, _message:String)
	{
		trace('Error! $_code : $_message');
	}

	static function onDisconnected(_code:Int, _message:String)
	{
		trace('Disconnected! $_code : $_message');
	}

	public static function initialize()
	{
		var DiscordDaemon = sys.thread.Thread.create(() ->
		{
			new DiscordClient();
		});
		trace("Discord Client initialized");
	}

	public static function changePresence(details:String, state:Null<String>, ?smallImageKey : String, ?hasStartTimestamp : Bool, ?endTimestamp: Float)
	{
		var startTimestamp:Float = if(hasStartTimestamp) Date.now().getTime() else 0;

		if (endTimestamp > 0)
		{
			endTimestamp = startTimestamp + endTimestamp;
		}

		DiscordRpc.presence({
			details: details,
			state: state,
			largeImageKey: 'icon',
			largeImageText: "fridaynightfunkin",
			smallImageKey : smallImageKey,
			// Obtained times are in milliseconds so they are divided so Discord can use it
			startTimestamp : Std.int(startTimestamp / 1000),
            endTimestamp : Std.int(endTimestamp / 1000)
		});

		//trace('Discord RPC Updated. Arguments: $details, $state, $smallImageKey, $hasStartTimestamp, $endTimestamp');
	}
}
#end

**Reply by A145918** [reply]
didn't work

**Comment by ghost** [comment]
Will your game build?
